src/Module.Server/Modes/Dtv/CrpgDtvServer.cs

```diff
@@ -36,15 +37,34 @@ internal class CrpgDTVServer : MissionMultiplayerGameModeBase
     public override bool AllowCustomPlayerBanners() => false;
     public override bool UseRoundController() => true;
 
-    public CrpgDTVServer(CrpgDTVClient dtvClient,
-        CrpgRewardServer rewardServer,
-        CrpgDTVTeamSelectComponent teamSelectComponent,
-        CrpgDTVSpawningBehavior spawningBehavior)
+    public CrpgDtvServer(CrpgDtvClient dtvClient,
+        CrpgRewardServer rewardServer)
     {
         _dtvClient = dtvClient;
         _rewardServer = rewardServer;
-        _teamSelectComponent = teamSelectComponent;
-        _spawningBehavior = spawningBehavior;
+        _dtvData = new XDocument();
+        _dtvData = XDocument.Load(ModuleHelper.GetXmlPath("Crpg", "dtv_data"));
+
+        if (_dtvData is not null)
+        {
+            try
+            {
+                totalWaves = _dtvData.Descendants("Rounds")
+                .Select(rounds => (int)rounds.Attribute("totalWaves"))
+                .FirstOrDefault();
+
+                totalRounds = _dtvData.Descendants("Rounds")
+                     .Select(rounds => (int)rounds.Attribute("totalRounds"))
+                     .FirstOrDefault();
+            }
+            catch (Exception e)
+            {
+                Debug.Print(e.Message);
+                totalRounds = 1;
+                totalWaves = 1;
+                Debug.Print("Failed to parse wave/round count from dtvData");
+            }
+        }
```

Comment:
Not sure if that deserialization logic should be located here. Anyway it would be nice to extract that in a method.

OPT: deserializing by hand the xml is not great. You should be able to generate a class with all the required attributes to deserialize with one line of code your xml.